hist/hist/src/TGraph.cxx

```diff
@@ -1058,158 +1058,89 @@ TObject *TGraph::FindObject(const TObject *obj) const
-///                                 Double_t *u, Int_t flag);
 ///
+/// This function is used for fitting also the derived TGraph classes such as TGraphErrors or TGraphAsymmErrors.
+/// See the note below on how the errors are used when fitting a TGraphErrors or TGraphAsymmErrors.
+/// 
+/// The fitting of the TGraph, i.e simple data points without any error associated, is performed using the 
+/// un-weighted least-square (chi-square) method.
+/// 
 ///
+///\anchor GFitErrors
 /// ### TGraphErrors fit:
 ///
-///   In case of a TGraphErrors object, when x errors are present, the error along x,
-///   is projected along the y-direction by calculating the function at the points x-exlow and
-///   x+exhigh. The chisquare is then computed as the sum of the quantity below at each point:
+///   In case of a TGraphErrors or TGraphAsymmErrors object, when `x` errors are present, the error along x,
+///   is projected along the y-direction by calculating the function at the points `x-ex_low` and
+///   `x+ex_high`, where `ex_low` and `ex_high` are the corresponding lower and upper error in x.
+///   The chi-square is then computed as the sum of the quantity below at each data point:
 ///
 /// \f[
 ///   \frac{(y-f(x))^{2}}{ey^{2}+(\frac{1}{2}(exl+exh)f'(x))^{2}}
 /// \f]
 ///
-///   where x and y are the point coordinates, and f'(x) is the derivative of the
-///   function f(x).
-///
-///   In case the function lies below (above) the data point, ey is ey_low (ey_high).
+///   where `x` and `y` are the point coordinates, and `f'(x)` is the derivative of the
+///   function `f(x)`.
 ///
-///   thanks to Andy Haas (haas@yahoo.com) for adding the case with TGraphAsymmErrors
-///             University of Washington
+///   In case of asymmetric errors, if the function lies below (above) the data point, `ey` is `ey_low` (`ey_high`).
 ///
 ///   The approach used to approximate the uncertainty in y because of the
 ///   errors in x is to make it equal the error in x times the slope of the line.
-///   The improvement, compared to the first method (f(x+ exhigh) - f(x-exlow))/2
-///   is of (error of x)**2 order. This approach is called "effective variance method".
-///   This improvement has been made in version 4.00/08 by Anna Kreshuk.
-///   The implementation is provided in the function FitUtil::EvaluateChi2Effective
+///   This approach is called "effective variance method" and 
+///   the implementation is provided in the function FitUtil::EvaluateChi2Effective
+///
+/// \anchor GFitLinear
+/// ### Linear fitting:
+///   When the fitting function is linear (contains the `++` sign) or the fitting
+///   function is a polynomial, a linear fitter is initialised.
+///   To create a linear function, use the following syntax: linear parts
+///   separated by `++` sign.
+///   Example: to fit the parameters of the function `p0*x + p1*sin(x)`, you can create a
+///   TF1 object as 
+///
+///       TF1 *f1 = new TF1("f1", "x++sin(x)", xmin, xmax);
```

Comment:
```suggestion
/// ~~~ {.cpp}
///       TF1 *f1 = new TF1("f1", "x++sin(x)", xmin, xmax);
/// ~~~
```
For code formatting

Reply:
You don't need the ~{.cpp}. It should do automatically if you have the right indentation